R/update.R

```diff
@@ -77,6 +83,55 @@ renv_update_find_github_impl <- function(record) {
 
 }
 
+renv_update_find_git_impl <- function(record) {
+
+  renv_git_preflight()
+
+  template <- c(
+    "git ls-remote \"${ORIGIN}\" \"${REF}\" | head -1 | sed 's/\t.*//'"
+  )
+
+  data <- list(
+    ORIGIN = record$RemoteUrl,
+    REF    = record$RemoteRef %||% record$RemoteSha
+  )
+
+  commands <- renv_template_replace(template, data)
+  command <- paste(commands, collapse = " && ")
+  if (renv_platform_windows())
+    command <- paste(comspec(), "/C", command)
+
+  sha <- system(command, intern = TRUE)
+
+  # if sha is empty, return NULL
+  # probably because recored$RemoteRef is SHA, not branch
+  if (length(sha) == 0) {
+    return(NULL)
+  }
+
+  # check for changed sha
+  if (sha == record$RemoteSha)
+    return(NULL)
```

Comment:
My feedback here:

Redundant logic - renv_retrieve_git_impl will handle this

Reply:
I'll refactor this into it's own function so its purpose (short-circuit cloning the repo if possible) is more obvious